### DISCUSSION TITLE
Using  Online GPS Tracker

### DISCUSSION THREAD

**Discussion by ramiro-alba** [osmandapp/OsmAnd#13476] [Q&A] [answered]
Hi all,

Trying to know  how 'Online GPS Tracker' works, I don't know how to use the parameter 'Web address'. Furthermore, the documentation about the matter does not fit with app. Can someone give some indications?

Thanks in advance,
Regards

**Comment by Lee-Carre** [comment]
This is likely for logging your track to a Web service. The `Web address` field is for entering a URL for it to upload data to.

**Reply by Lee-Carre** [reply]
For reference, the example URL given in the UI is

    https://example.com?lat={0}&lon={1}&timestamp={2}&hdop={3}&altitude={4}&speed={5}

additionally the descriptive hint text states that

    bearing={6}

So, in summary of the parameters:
* {0} = latitude (presumably in decimal degrees North)
* {1} = longitude (presumably in decimal degrees East)
* {2} = timestamp (I would hope in [ISO 8601 notation](//www.cl.cam.ac.uk/~mgk25/iso-time.html))
* {3} = HDoP
* {4} = altitude (I would hope above the geoid (MSL))
* {5} = speed
* {6} = bearing

**Comment by ramiro-alba** [comment]
Lee,

So you need to configure a web server in order to receive pushes from
osmand, do you?
But what about sharing location using telegram?

Regards

**Reply by Lee-Carre** [reply]
Hmm, this should've been posted as a reply to my answer, rather than as another answer. I guess GitHub does the simplest thing when replying directly via e-mail.

> So you need to configure a web server in order to receive pushes from osmand, do you?

I gather so. Or use someone else's.

This is likely by (wise) design; HTTP is generic / flexible, common (damn near ubiquitous), & (relatively) simple.

At least, that's how I'd want it arranged.

> But what about sharing location using telegram?

What of it? That's Telegram's problem, not OsmAnd's. There already exist mechanisms which involve simple transfer (either copy & paste, or via IPC intents) of textual strings (e.g. geo: URIs).

Last I checked, Telegram is non-libre. Proprietary systems tend to not play well with others.

However, with that being said …

Directly (in a way which some would call ‘seamless’ or integrated)? Well, I would assume that a different protocol is used (I don't use Telegram, so care to cite the specs?). Can't expect OsmAnd (or any app) to implement all protocols for what is essentially a nice-to-have feature.

Plus, OsmAnd would need a specific end-point (hostname). Depending how Telegram does things, that may not exist. Because, besides getting the data out of OsmAnd, the next problem is access-control. Which (Telegram) users receive which messages, is an internal matter of Telegram. Telegram's ‘solution’ may well be for OsmAnd to implement the entire protocol to become a Telegram client (which is ridiculous).

The *modular* way would be to do the conversion on the backend, from receiving HTTPd to whatever & wherever Telegram expects. That keeps OsmAnd (and each other component) simpler: [POSIX principles](http://www.catb.org/esr/writings/taoup/).

It also depends on the intent of Telegram's position-sharing;
* single fix, akin to a regular message, to reveal where you are right now, or where the meeting place will be, as a one-off
* continually being able to track the movements of friends; recurring updates (a new fix every N seconds)

Those two are significantly different.

For the former, you could just send a geo:[…] in a regular conversational message (just as you would a URL, or any other URI). The recipient can open that in software of their choosing.

If Telegram has some kind of special message type (for locations) & it were well designed & implemented, then it should be able to (optionally) auto-convert a geo: URI into its own special message type. I wouldn't count on that being available; perhaps submit a bug report if there isn't one already.

For the latter (recurring updates); that's more complicated, and tends to be better achieved with dedicated systems (hence this realtime-export function within OsmAnd's plugin, to feed into such a system from OsmAnd (instead of needing a separate client app running)), which tend to be based on HTTP anyway, from what I've seen. Especially if you want the sharing to be 2-way (telling your friends where you are, and being able to see where they are).

Beyond that, it depends too much on the contextual specifics. Describe the scenario and what you're trying to achieve ([other hints & tips](/selfteaching/How-To-Ask-Questions-The-Smart-Way/blob/master/How-To-Ask-Questions-The-Smart-Way.md)).

**Reply by Lee-Carre** [reply]
Having done some searching & reading, we may be discussing two entirely different things.

I was referring to the feature of the Trip Recording plugin.

However, there seems to be something very different with a similar name: [OsmAnd Online GPS Tracker](//osmand.net/features/tracker). Ironically, this seems to be intended for Telegram fans.

**Comment by ramiro-alba** [comment, marked as answer]
Lee,

Very interesting your discussion. Not specially interested in using OsmAnd
with Telegram, just doing research with the possibilities of OsmAnd, which
I recently discovered and find incredible. Many thanks for your answer

Regards

**Comment by NickVeld** [comment]
If you want to push location to Telegram somehow, you can use the following app: https://github.com/NickVeld/geo_online_tracking_to_messenger

It works as a local web server on a Android device and send links to OpenStreetMaps with obtained coordinates to Telegram.
The app is limited now but if there is a request, new things can be added.
